cobbler/utils/event.py

```diff
@@ -0,0 +1,72 @@
+"""
+This module contains logic to support the events Cobbler generates in its XML-RPC API.
+"""
+
+import time
+import uuid
+
+from cobbler import enums
+
+
+class CobblerEvent:
+    """
+    This is a small helper class that represents an event in Cobbler.
+    """
+
+    def __init__(self, name="", statetime=0.0):
+        """
+        Default Constructor that initializes the event id.
+
+        :param name: The human-readable name of the event
+        :statetime: The time the event was created.
+        """
+        self.__event_id = ""
+        self.statetime = statetime
+        self.__name = name
+        self.state = enums.EventStatus.INFO
+        self.read_by_who = []
+        # Initialize the even_id
+        self.__generate_event_id()
+
+    @property
+    def event_id(self):
+        """
+        Read only property to retrieve the internal ID of the event.
+        """
+        return self.__event_id
+
+    @property
+    def name(self):
+        """
+        Read only property to retrieve the human-readable name of the event.
+        """
+        return self.__name
+
+    def __generate_event_id(self):
+        """
+        Generate an event id based on the current timestamp
+
+        :return: An id in the format: "<4 digit year>-<2 digit month>-<two digit day>_<2 digit hour><2 digit minute>
+                 <2 digit second>_<optional string>"
+        """
+        (
+            year,
+            month,
+            day,
+            hour,
+            minute,
+            second,
+            weekday,
+            julian,
+            dst,
+        ) = time.localtime()
+        task_uuid = uuid.uuid4().hex
+        self.__event_id = f"{year:04d}-{month:02d}-{day:02d}_{hour:02d}{minute:02d}{second:02d}_{self.name}_{task_uuid}"
+
+    def to_list(self) -> list:
```

Comment:
This method is not needed and an anti-pattern. I'd make this class iterable and use `list()` on it to create a list. All that's needed is `__len__` and `__getitem__`. One way to implement them is:
```python
class CobblerEvent:
    def __init__(self, name="", statetime=0.0):
        ...
        # HACK: Due to @property it's easier to list the members than work with inspection
        self.members = [self.statetime, self.name, self.state.value, self.read_by_who]

    def __len__(self):
        return len(self.members)

    def __getitem__(self, idx):
        return self.members[idx]
```

And then you get the same, but you take advantage of Python's data model:
```python
>>> c = CobblerEvent("dummy")
>>> list(c) == c.to_list()
True
>>> list(c)
[0.0, 'dummy', 'info', []]
```

Reply:
Done. Thanks for the suggestion. It seems I have some catching up to do what pythonic means. I learned a lot of anti-patterns from this repository in the past it seems... :)